Cython/Compiler/PyrexTypes.py

```diff
@@ -4118,6 +4124,72 @@ def same_as_resolved_type(self, other_type):
     def error_condition(self, result_code):
         return "dummy"
 
+class InitOrClassVar: #(PyrexType):
```

Comment:
Just highlighting this because: I don't think this is awful but I'm not sure all the details are quite right. It's probably worth getting right because if done well you could implement PEP-484 largely using the same model.